app/code/core/Mage/Adminhtml/controllers/Permissions/RoleController.php

```diff
@@ -232,30 +225,35 @@ public function saveRoleAction()
                 ->setResources($resource)
                 ->saveRel();
 
-            foreach($oldRoleUsers as $oUid) {
+            foreach ($oldRoleUsers as $oUid) {
                 $this->_deleteUserFromRole($oUid, $role->getId());
             }
 
             foreach ($roleUsers as $nRuid) {
                 $this->_addUserToRole($nRuid, $role->getId());
             }
 
-            $rid = $role->getId();
+            $users = Mage::getResourceModel('admin/user_collection')
+                ->addFieldToFilter('user_id', ['in' => Mage::getModel('admin/roles')->setId($role->getId())->getRoleUsers()]);
+
+            foreach ($users as $user) {
```

Comment:
in \Mage_Admin_Model_Resource_Roles::_updateRoleUsersAcl there is a more performant way to update the flag for a big list of users.
This may work better for Shops which may have a lot of Admin Users.


https://github.com/OpenMage/magento-lts/blob/f71321b37a782bd167ef56de0c6e18e67645685d/app/code/core/Mage/Admin/Model/Resource/Roles.php#L166-L179

not sure if its better to open up this method, or to copy the code into a private method here. Probably the Copy for now.

Reply:
You are so right, moreover, my `foreach` is so useless, because the previous `$role->save()` run `_updateRoleUsersAcl`. I updating and testing again.